core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -620,6 +620,14 @@ private[spark] class TaskSetManager(
     // Note: "result.value()" only deserializes the value when it's called at the first time, so
     // here "result.value()" just returns the value and won't block other threads.
     sched.dagScheduler.taskEnded(tasks(index), Success, result.value(), result.accumUpdates, info)
+    // Kill other task attempts if any as the one attempt succeeded
+    for (attemptInfo <- taskAttempts(index) if attemptInfo.attemptNumber != info.attemptNumber
```

Comment:
Thanks @tgravescs.

I would be happy to fix the issue about succeeding more than one attempt as you explained as part of this PR but I am thinking it would be good if we can handle it separately without mixing with the current PR changes.

I will move the current changes to a PR for [SPARK-10530](https://issues.apache.org/jira/browse/SPARK-10530) and we can continue to fix multiple attempts success issue as part of the [SPARK-13343](https://issues.apache.org/jira/browse/SPARK-13343). 
Please let me if it doesn't make sense.